pkg/api/validation/validation.go

```diff
@@ -3517,3 +3530,17 @@ func isValidHostnamesMap(serializedPodHostNames string) bool {
 	}
 	return true
 }
+
+func sysctlIntersection(a []api.Sysctl, b []api.Sysctl) []api.Sysctl {
+	lookup := make(map[string]struct{}, len(a))
+	result := []api.Sysctl{}
+	for i := range a {
+		lookup[a[i].Name] = struct{}{}
+	}
+	for i := range b {
+		if _, found := lookup[b[i].Name]; found {
+			result = append(result, b[i])
+		}
+	}
+	return result
+}
```

Comment:
I think you dont have to store whole sysctl into result if you are just using their name in the validation, see my other comment above.